app/routes/app+/market_.$address.submission.$submissionId.tsx

```diff
@@ -73,19 +72,20 @@ export const loader = async (data: DataFunctionArgs) => {
   const serviceRequest = await findServiceRequest(submission.serviceRequestId, address);
   invariant(serviceRequest, "Service request not found");
 
-  return typedjson({ submission, reviews, params, laborMarket, reviewedByUser, serviceRequest }, { status: 200 });
+  return typedjson({ submission, reviews, params, laborMarket, user, serviceRequest }, { status: 200 });
 };
 
 const reviewSubmissionMachine = createBlockchainTransactionStateMachine<ReviewContract>();
 
 export default function ChallengeSubmission() {
-  const { submission, reviews, params, laborMarket, reviewedByUser, serviceRequest } =
-    useTypedLoaderData<typeof loader>();
+  const { submission, reviews, params, laborMarket, user, serviceRequest } = useTypedLoaderData<typeof loader>();
   const submit = useSubmit();
   const formRef = useRef<HTMLFormElement>(null);
   const mType = laborMarket.appData?.type;
   invariant(mType, "Labor Market type is undefined");
   const [searchParams, setSearchParams] = useSearchParams();
+  const reviewedByUser = user && reviews.find((review) => review.reviewer === user.address);
```

Comment:
Going to add a ticket to move this to the backend. Otherwise this only checks the 10 reviews on that page and not all of them.

Reply:
That's what it is supposed to do? Only checks to see if this submission was reviewed by the current user

Reply:
The `reviews` here comes from `searchReviews` which only pulls 10 at a time. Say there's 100 reviews, this client-side logic breaks.

Reply:
Oh ok, gotcha